s3transfer/download.py

```diff
@@ -79,6 +83,27 @@ def get_fileobj_for_io_writes(self, transfer_future):
         """
         raise NotImplementedError('must implement get_fileobj_for_io_writes()')
 
+    def queue_file_io_task(self, fileobj, data, offset):
+        """Queue IO write for submission to the IO executor.
+
+        This method accepts an IO executor and information about the
+        downloaded data, and handles submitting this to the IO executor.
+
+        This method may defer submission to the IO executor if necessary.
+
+        """
```

Comment:
Ideally I wanted to extract the `_submit_task` into a separate class so other things besides a Task could submit things.  Would giving the transfer coordinator the ability to submit tasks make sense?  We'd then update `_submit_task` to just go through the transfer coordinator.